subiquity/models/tests/test_filesystem.py

```diff
@@ -78,3 +159,702 @@ def test_errors(self):
+        self.assertTrue(disk.available)
+
+    def test_disk_not_available(self):
+        """verify disk is is False with disk.used >= disk.size"""
+        disk = Disk.from_info(self.info)
+        disk._partitions.append(Partition(device=disk, size=disk.size))
+        self.assertFalse(disk.available)
+
+
+class TestFilesystemPartition(CiTestCase):
+    def test_partition_filesystem(self):
+        """verify partition.fs() returns the filesystem object added"""
+        part = Partition()
+        fs = Filesystem()
+        fs.fstype = 'ext4'
+        fs.volume = part
+        fs.label = 'cloudimg-rootfs'
+        part._fs = fs
+        self.assertEqual(fs, part.fs())
+
+    def test_partition_desc(self):
+        """verify partition.desc() returns description combined from device"""
+        disk = Disk()
+        part = Partition()
+        part.device = disk
+        self.assertEqual('partition of local disk', part.desc())
+
+    def test_partition_available(self):
+        """verify by default partitions are available"""
+        partition = Partition()
+        self.assertTrue(partition.available)
+
+    def test_partition_not_available_bios_grub(self):
+        """partition.available is False when partition.flag is bios_grub"""
+        partition = Partition()
+        partition.flag = "bios_grub"
+        self.assertFalse(partition.available)
+
+    def test_partition_not_available_has_fs(self):
+        """partition.available is False with partition has a fs association"""
+        partition = Partition()
+        partition._fs = Filesystem()
+        self.assertFalse(partition.available)
+
+    def test_partition_available_delegates_to_fs_mount(self):
+        """partition.available delegates to fs._mount"""
+        def _fsobjs():
+            """ Extract the valid FS entries in
+                FilesystemModel.supported_filesystems list """
+            return [fstup for fstup in FilesystemModel.supported_filesystems
+                    if len(fstup) > 2 and fstup[2].label]
+
+        for (fsname, _disp, fs_obj) in _fsobjs():
+            partition = Partition()
+            fs = Filesystem()
+            fs.fstype = fsname
+            partition._fs = fs
+            self.assertEqual(fs_obj.is_mounted, partition.available)
+
+    def test_partition_not_available_if_mounted(self):
+        """partition.available is False with mounted"""
+        partition = Partition()
+        partition._fs = Filesystem()
+        partition._fs._mount = "Not None"
+        self.assertFalse(partition.available)
+
+    def test_partition_number(self):
+        """partition.number derived from device partition index"""
+        disk = Disk()
+        partition = Partition()
+        partition.device = disk
+        disk._partitions.append(partition)
+        self.assertEqual(1, partition._number)
+
+    # FIXME: partition.path does not account for named partition (nvmen0p1)
+    def test_partition_path(self):
```

Comment:
I wonder if we actually display the partition's path anywhere. We probably shouldn't.